charts/tekton-apps/Chart.yaml

```diff
@@ -186,6 +186,7 @@ description: |
     - apps[PROJECT].components[NAME].argocd.source.targetRevision - tag or branch in the repository for ArgoCD Application (default: kubernetes branch for basic projects
       "<apps[PROJECT].kubernetesRepository.branch>" or "11.0.14" for wordpress projects)
     - apps[PROJECT].components[NAME].applicationURL - url that should be used in tekton build ConfigMap `APPLICATION_URL` param
+    - apps[PROJECT].components[NAME].ignoreDeploymentReplicasDiff - flag whether this exact ArgoCD application should ignore `Replicas` count differences for deployments. It may be needed for `staging` and `prod` environments which use HPA (default: false)
```

Comment:
Maybe better to set a default to true? It is a best practice to not use GitOps for replica's count while using HPA
https://argo-cd.readthedocs.io/en/stable/user-guide/best_practices/#leaving-room-for-imperativeness

Reply:
> cause we have a lot of applications and components in dev env for which I don't think we should turn on this possibility to ignore replicas differences.
Why? How often do you manipulate with replicas count on dev? For me in 99% of time it's just 1.

My logic is that impact of setting wrong value on prod is much higher, and may be we should use prod-oriented default values in charts and override them on dev/staging, not vice versa. But adding this to boilerplate works for me too.

Reply:
1. for dev envs we should never have any HPAs, we can test multi-pod by manually set in dev env kustomize patch (replicas: 2) if we need to. 
2. for staging and prod envs HPA is the default way going forward
3. @roman-mightydevops `Explicit is better than implicit.` - so that means that I prefer a statement in values.yaml saying we "IGNORE" the diff for certain values (explicit) VS hiding it with auto assumption (implicit). So @kseniyashaydurova had a correct implementation (as she comes with Python's ZEN: https://peps.python.org/pep-0020/)